lib/mixlib/install/generator/base.rb

```diff
@@ -48,6 +48,22 @@ def self.get_script(name, context = {})
             # Default values to use incase they are not set in the context
             context[:base_url] ||= "https://omnitruck.chef.io"
 
+            require "mixlib/install/version"
+            mixlib_install_agent = "mixlib-install/#{Mixlib::Install::VERSION}"
+            # Ensure default mixlib-install agent header is included
+            context[:user_agent_string] = if context[:user_agent_string]
+                                            if context[:user_agent_string].include?(mixlib_install_agent)
+                                              # Pass the user_agent_string along if it contains the default agent
+                                              context[:user_agent_string]
+                                            else
+                                              # Prepend the default agent if it doesn't exist
+                                              "#{mixlib_install_agent} #{context[:user_agent_string]}"
+                                            end
+                                          else
+                                            # Set the default agent if user_agent_string is not set
+                                            mixlib_install_agent
+                                          end
+
```

Comment:
Didn't work. Part of why it's so complex is not to duplicate the default header. But, I'll work off of this solution.